src/amplitude.js

```diff
@@ -319,10 +343,38 @@ Amplitude.prototype.setVersionName = function(versionName) {
   }
 };
 
+// truncate string values in event and user properties so that request size does not get too large
+Amplitude.prototype._truncate = function(value) {
+  if (typeof(value) === 'object') {
```

Comment:
typeof(new String()) === 'object'
instead of rewriting these, we should use a library like https://github.com/component/type